lib/features/payment/payment_details_page.dart

```diff
@@ -23,49 +23,47 @@ import 'package:tbdex/tbdex.dart';
 
 class PaymentDetailsPage extends HookConsumerWidget {
   final PaymentState paymentState;
+  final PaymentDetailsState paymentDetailsState;
 
   const PaymentDetailsPage({
     required this.paymentState,
+    required this.paymentDetailsState,
     super.key,
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final selectedPaymentMethod = useState<Object?>(null);
+    final selectedPaymentMethod = useState<PaymentMethod?>(null);
     final selectedPaymentType = useState<String?>(null);
     final offeringCredentials = useState<List<String>?>(null);
     final rfq = useState<AsyncValue<Rfq>?>(null);
 
-    final paymentMethods = _getPaymentMethods(paymentState);
-    final paymentTypes = _getPaymentTypes(paymentMethods);
-    final filteredPaymentMethods =
-        _getFilteredPaymentMethods(paymentMethods, selectedPaymentType.value);
+    final availableMethods =
```

Comment:
Another option here (and I don't know if it's better) would be to just have a single `PaymentDetailsState` that is used for all these substates. Then you would just have 
```dart
final state = useState<PaymentDetailsState>(paymentDetailsState)
```
Then you could observe that where needed. But tbh that might be a bit annoying and would require copyWith for each of these sub states. It also could be less efficient since it would rebuild the whole pages on state changes. But tbh, so do these states :)